### PR TITLE
completed feature user-can-view-detail-for-a-single-recipe - Mohamed 

### PR DESCRIPTION
**Pull request by Mohamed-Elmessiry**
Screen Shot 2021-04-19 at 12 17 54 PM
Screen Shot 2021-04-19 at 12 18 28 PM

### CODE REVIEW

**Comment by Mohamed-Elmessiry** [comment]
Screen Shot 2021-04-22 at 6 23 35 PM
Screen Shot 2021-04-22 at 6 23 39 PM
Screen Shot 2021-04-22 at 6 27 46 PM

File: css/styles.css [hunk 1]
```diff
@@ -113,6 +116,82 @@ h1 {
...
+  border: none;
+  width: 118px;
+  height: 33px;
+  margin-right: 100px;
+  border-radius: 5px;
+}
+
+.list-holder {
+  line-height: 3;
+  margin-left: 336px;
```
Comment (codingcodymiller) [inline css/styles.css]: This is a lot of margin. Why is this necessary? Are you trying to center this element in the page?

File: css/styles.css [hunk 2]
```diff
@@ -113,6 +116,82 @@ h1 {
...
+  width: 100%;
+}
+
+.ingredients-heading {
+  font-family: 'Anton', sans-serif;
+  color: #4b4b4b;
+  margin-right: 650px;
```
Comment (codingcodymiller) [inline css/styles.css]: 😱 😱 😱 😱 😱

File: css/styles.css [hunk 3]
```diff
@@ -113,6 +116,82 @@ h1 {
...
+}
+
+.list-element {
+  font-size: 10px;
+  font-family: 'andika', sans-serif;
+  margin-right: 870px;
```
Comment (codingcodymiller) [inline css/styles.css]: 😱 😱 😱 😱 😱

File: js/main.js [hunk 4]
```diff
@@ -54,11 +56,66 @@ function createElement(obj) {
...
   photoDiv.style.backgroundImage = 'url(' + obj.recipe.image + ')';
   photoDiv.setAttribute('class', 'results-photo-div');
+  photoDiv.setAttribute('id', obj.recipe.label);
   resultHolder.appendChild(photoDiv);
   var dishName = document.createElement('h1');
+  dishName.setAttribute('class', 'dish-name');
   dishName.innerHTML = obj.recipe.label;
   resultHolder.appendChild(dishName);
 
   resultsContainer.appendChild(resultHolder);
```
Comment (codingcodymiller) [inline js/main.js]: You shouldn't be appending a dynamically created DOM tree directly to the DOM, you should return that DOM tree from the function, and append it where you want from where the function is called. I would review `dom-creation` for a reminder on how to properly generate DOM trees for a set of data.

File: js/main.js [hunk 5]
```diff
@@ -54,11 +56,66 @@ function createElement(obj) {
...
+        saveButton.textContent = 'Save to Favourties';
+        saveButton.className = 'save-to-favorites';
+        buttonDiv.appendChild(saveButton);
+
+        singleResultDiv.appendChild(singleResultHolder);
+      }
+    }
+
+  }
+});
```
Comment (codingcodymiller) [inline js/main.js]: Why is this function an anonymous function? That is a lot of code, there is no doubt a name you could give this function to describe what it is meant to do.

File: js/main.js [hunk 6]
```diff
@@ -54,11 +56,66 @@ function createElement(obj) {
...
+        var buttonDiv = document.createElement('div');
+        buttonDiv.setAttribute('class', 'buttons-holder');
+        singleResultHolder.appendChild(buttonDiv);
+        var fullRecipeButton = document.createElement('button');
+        fullRecipeButton.value = 'View Full Recipe';
+        fullRecipeButton.textContent = 'View Full Recipe';
+        fullRecipeButton.className = 'view-full-recipe';
+        fullRecipeButton.addEventListener('click', function () {
+          document.location = dataArray.hits[i].recipe.url;
+        });
```
Comment (codingcodymiller) [inline js/main.js]: Are you using javascript to redirect the user to a new page? There is an HTML element for that purpose: `a`

File: js/main.js [hunk 7]
```diff
@@ -54,11 +56,66 @@ function createElement(obj) {
...
+        buttonDiv.appendChild(fullRecipeButton);
+        var saveButton = document.createElement('button');
+        saveButton.textContent = 'Save to Favourties';
+        saveButton.className = 'save-to-favorites';
+        buttonDiv.appendChild(saveButton);
+
+        singleResultDiv.appendChild(singleResultHolder);
```
Comment (codingcodymiller) [inline js/main.js]: This DOM tree is again being appended directly to the page inside of the function. You should be passing a function the data it needs to generate a DOM tree, and then returning the DOM tree for that data, no affecting the DOM directly from within the function.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: css/styles.css (hunk 1 above)
Comment (Mohamed-Elmessiry) [reply css/styles.css:145]: It's beacause they are not centered in the wireframes, so I am trying to adjust them to have the same identical as possible

File: css/styles.css (hunk 2 above)
Comment (Mohamed-Elmessiry) [reply css/styles.css]: it's because they are not centred in the wireframe, so I am trying to adjust them to look the same

**Comment by Mohamed-Elmessiry** [comment]
Screen Shot 2021-04-23 at 2 33 42 PM
Screen Shot 2021-04-23 at 2 33 44 PM
Screen Shot 2021-04-23 at 2 33 54 PM
Screen Shot 2021-04-23 at 2 33 59 PM

**Review by codingcodymiller — APPROVED** [review]
